[DPE-7914] Clarify how to use self-signed S3 endpoints (#164)
canonical/spark-k8s-bundle: docs/how-to/self-signed-certificates.md

(how-to-self-signed-certificates)=

# Support self-signed certificates in Charmed Apache Spark

In some use cases, there is the need to use self-signed certificates to trust self-hosted services such as Ceph or many others.

> ## Spark-client snap

Reviewer: Suggestion: Since it's a how-to guide, I'd prefer to have more imperative headings and steps.

The Spark-client snap offers the possibility to submit jobs to a Kubernetes cluster with the `spark-submit` command.

> For this reason, we have a feature in the Apache Spark Client snap to add certificates in the Java truststore inside the snap to validate the desired service.

Author: Yes, this makes sense. Since the sentence was spark-client specific, it felt weird to have it as an introduction to the whole document, but I'll rephrase it.

To add a new certificate you can use the following command:

```bash
spark-client.import-certificate <CERTIFICATE_ALIAS> <CERTIFICATE_PATH>
```

where `<CERTIFICATE_ALIAS>` is the alias associated to the certificate and the `<CERTIFICATE_PATH>` is the path of the desired certificate. Please be sure that the path of the certificate is accessible by the Apache Spark Client snap.

For more information, see the blog post on [how to deploy Charmed Apache Spark, with MAAS, Kubernetes and Ceph with self-signed certificates](https://ubuntu.com/blog/deploy-an-on-premise-data-hub-with-canonical-maas-spark-kubernetes-and-ceph).

> ## Apache Spark applications

Reviewer: Suggestion: Can we make this heading more imperative, clearly stating what action we are trying to perform?

(how-to-self-signed-certificates)=

# Support self-signed certificates in Charmed Apache Spark

In some use cases, there is the need to use self-signed certificates to trust self-hosted services such as Ceph or many others.
For this reason, both the Spark-client snap and the Apache Spark applications need to add certificates in their respective truststore to validate the desired service.

## Import certificates in the Spark-client snap

To add a new certificate you can use the following command:

```bash
spark-client.import-certificate <CERTIFICATE_ALIAS> <CERTIFICATE_PATH>
```

> where `<CERTIFICATE_ALIAS>` is the alias associated to the certificate and the `<CERTIFICATE_PATH>` is the path of the desired certificate. Please be sure that the path of the certificate is accessible by the Apache Spark Client snap.

Reviewer: Suggestion: Let's remove the placeholder explanation, as it's explaining the obvious part.

```suggestion
Please make sure that the path of the certificate is accessible by the Apache Spark Client snap.
```
